[develop] Fail build image when using a kernel version not supported by FSx for Lustre

enrico-usai opened the pull request:
Initially we decided to not fail, because when we released the first version of RH8 support only 8.4 was available, so we didn't want to release a broken functionality.

Anyway at that point we can use the same approach we're using for CentOS7 and fail the build if the kernel version is not compatible with FSx for Lustre.

This will permit us and users to discover this incompatibility at build image time, rather than at cluster creation time if using FSx storage.

### Tests
Verified with an AMI with kernel `4.18.0-425.13.1.el8_7.x86_64`

```
bash kitchen.ec2.sh environment-install converge lustre-rhel8
...
[2023-11-22T15:13:09+00:00] FATAL: RuntimeError: test_resource[resource] (aws-parallelcluster-tests::test_resource line 30) had an error:
RuntimeError: lustre[test] (aws-parallelcluster-tests::test_resource line 8) had an error:
RuntimeError: FSx for Lustre is not supported in kernel version 4.18.0-425.13.1.el8_7.x86_64 of RHEL 8.7, please update the kernel version
```

Commit:
Fail build image when using a kernel version not supported by FSx for Lustre

Initially we decided to not fail, because when we released the first version of RH8
support only 8.4 was available, so we didn't want to release a broken functionality.

Anyway at that point we can use the same approach we're using for CentOS7
and fail the build if the kernel version is not compatible with FSx for Lustre.

This will permit us and users to discover this incompatibility at build image time,
rather than at cluster creation time if using FSx storage.

### Tests
Verified with an AMI with kernel `4.18.0-425.13.1.el8_7.x86_64`

```
bash kitchen.ec2.sh environment-install converge lustre-rhel8
...
[2023-11-22T15:13:09+00:00] FATAL: RuntimeError: test_resource[resource] (aws-parallelcluster-tests::test_resource line 30) had an error:
RuntimeError: lustre[test] (aws-parallelcluster-tests::test_resource line 8) had an error:
RuntimeError: FSx for Lustre is not supported in kernel version 4.18.0-425.13.1.el8_7.x86_64 of RHEL 8.7, please update the kernel version
```

Enable `tag:install` kitchen test to fail if `lnet is not correctly installed.

Signed-off-by: Enrico Usai <usai@amazon.com>

jdeamicis commented on cookbooks/aws-parallelcluster-environment/spec/unit/resources/lustre_setup_spec.rb:
```diff
@@ -185,8 +187,10 @@ def self.setup(chef_run)
-          .with(level: :warn)
+        expect { chef_run }.to(raise_error do |error|
+          expect(error).to be_a(Exception)
+          expect(error.message).to include("FSx for Lustre is not supported in kernel version #{kernel_version} of RHEL #{platform_version}, please update the kernel version")
```
Isn't there a similar test for Rocky? Maybe it's not really necessary though...

enrico-usai replied:
Good point, we missed this test for rocky

enrico-usai replied:
added

Commit:
Add ChefSpec test coverage for Rocky Linux lustre installation

+ Align RHEL and Rocky Linux messages
+ Add coverage for version 513 of the kernel

Signed-off-by: Enrico Usai <usai@amazon.com>
